modules/guardduty/remote-state.tf

```diff
@@ -0,0 +1,29 @@
+module "account_map" {
+  source  = "cloudposse/stack-config/yaml//modules/remote-state"
+  version = "1.4.2"
+
+  component   = "account-map"
+  tenant      = (var.account_map_tenant != "") ? var.account_map_tenant : module.this.tenant
+  stage       = var.root_account_stage
+  environment = var.global_environment
+  privileged  = var.privileged
+
+  context = module.this.context
+}
+
+module "guardduty_delegated_detector" {
+  source  = "cloudposse/stack-config/yaml//modules/remote-state"
+  version = "1.4.2"
+
+  # If we are creating the delegated detector (because we are in the delegated admin account), then don't try to lookup
+  # the delegated detector ID from remote state
+  count = local.create_guardduty_collector ? 0 : 1
+
+  component   = "${var.delegated_admininstrator_component_name}/${module.this.environment}"
+  tenant      = module.this.tenant
+  stage       = replace(var.delegated_administrator_account_name, "${module.this.tenant}-", "")
```

Comment:
Yes, because we are getting the name of the stage that is the delegated administrator as `core-security`, and we need to reference this both from the `root` account and the `core-security` account itself, depending on how the component is being used, so we can't just rely on the context.tf.